blocklib/blocks/copy/copy_cpu.hpp

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include <gnuradio/blocks/copy.hpp>
+
+namespace gr {
+namespace blocks {
+
+class copy_cpu : public copy
+{
+public:
+    copy_cpu(size_t itemsize) : copy(itemsize) {}
+    virtual work_return_code_t work(std::vector<block_work_input>& work_input,
+                                    std::vector<block_work_output>& work_output) override;
+
+protected:
+    size_t d_itemsize;
+};
+
+// #define PARAM_LIST size_t itemsize
```

Comment:
Surely not `#define`? What is this, C++98? :-)

Reply:
Is there something other than `#define` or some pattern I'm not aware of that would achieve the same purpose - literally replacing the list of params.  Was considering this for the case where the param list is autogenerated, but then needs to be used also in the manually edited `.cpp` files so that list could be dynamically driven by the yaml.

Reply:
Hmm… yeah now looking at how it's actually used.
Maybe the code can be generated? It's only really for the `make` function?

I think using macros here probably will complicate a future move to C++ modules, too.

Reply:
Would it be better here just to encapsulate the args in a struct, then pass that around.  Then the whole list of args doesn't have to be repeated over and over - especially when the code isn't or can't be generated such as specialized template constructors or something
```
class copy : public sync_block
{
public:
    typedef struct 
    {
        size_t itemsize;
    } block_args;

    enum class available_impl {CPU, CUDA};

    typedef std::shared_ptr<copy> sptr;
    copy(size_t itemsize) : sync_block("copy")
    {
        add_port(untyped_port::make("in", port_direction_t::INPUT, itemsize));
        add_port(untyped_port::make("out", port_direction_t::OUTPUT, itemsize));
    }

    static sptr make(block_args args, available_impl impl = available_impl::CPU);
```